lib/ui/form/widget/form_survey_answer_text_field.dart

```diff
@@ -1,64 +1,67 @@
+import 'package:flutter_survey/model/submit_survey_question_model.dart';
 import 'package:flutter_survey/theme/app_dimensions.dart';
 import 'package:flutter_survey/ui/widget/custom_text_field.dart';
-import 'package:collection/collection.dart';
 
 class FormSurveyAnswerTextField extends StatefulWidget {
-  final QuestionModel question;
-  const FormSurveyAnswerTextField({super.key, required this.question});
+  final List<AnswerModel> answers;
+  final ValueChanged<List<SubmitSurveyAnswerModel>> onUpdateAnswer;
+
+  const FormSurveyAnswerTextField({
+    super.key,
+    required this.answers,
+    required this.onUpdateAnswer,
+  });
 
   @override
   State<FormSurveyAnswerTextField> createState() =>
       _FormSurveyAnswerTextFieldState();
 }
 
 class _FormSurveyAnswerTextFieldState extends State<FormSurveyAnswerTextField> {
-  final List<TextEditingController> _controllers = [];
-  late List<String> _textFieldHints = [];
+  late List<SubmitSurveyAnswerModel> _answerModels = [];
 
   @override
   void initState() {
     super.initState();
-    _textFieldHints =
-        widget.question.answers.map((element) => element.text).toList()
-          ..forEach((element) {
-            _controllers.add(TextEditingController());
-          });
+    _answerModels = widget.answers
+        .map((element) => SubmitSurveyAnswerModel(
+              id: element.id,
+            ))
+        .toList();
   }
 
   @override
   Widget build(BuildContext context) {
     return SingleChildScrollView(
       child: Column(
         mainAxisAlignment: MainAxisAlignment.center,
-        children: [
-          for (final textAndControllerPair in IterableZip([
-            _textFieldHints,
-            _controllers,
-          ]))
-            Padding(
-              padding: const EdgeInsets.only(bottom: AppDimensions.spacing20),
-              child: customTextField(
-                context: context,
-                controller: textAndControllerPair[1] as TextEditingController,
-                textInputType: textAndControllerPair[0] as String == 'Email'
-                    ? TextInputType.emailAddress
-                    : TextInputType.text,
-                isObscuredText: false,
-                hintText: textAndControllerPair[0] as String,
+        children: widget.answers
+            .mapIndexed(
+              (index, element) => Padding(
+                padding: const EdgeInsets.only(bottom: AppDimensions.spacing20),
+                child: customTextField(
+                  context: context,
+                  controller: null,
+                  textInputType: widget.answers[index].text == 'Email'
+                      ? TextInputType.emailAddress
+                      : TextInputType.text,
+                  isObscuredText: false,
+                  hintText: widget.answers[index].text,
+                  onChanged: (text) {
+                    _answerModels[index] = SubmitSurveyAnswerModel(
+                      id: _answerModels[index].id,
```

Comment:
Something did not match with [my suggestion](https://github.com/nimblehq/ic-flutter-taher-toby/pull/90#discussion_r1192112053):
- the element (answer) was not used to replace `widget.answers[index]` and `_answerModels[index]`.

What do you think?

```suggestion
              (index, answer) => Padding(
                padding: const EdgeInsets.only(bottom: AppDimensions.spacing20),
                child: customTextField(
                  context: context,
                  controller: null,
                  textInputType: answer.text == 'Email'
                      ? TextInputType.emailAddress
                      : TextInputType.text,
                  isObscuredText: false,
                  hintText: answer.text,
                  onChanged: (text) {
                    _answerModels[index] = SubmitSurveyAnswerModel(
                      id: answer.id,
```